app/controllers/declaration/DeclarationHolderChangeController.scala

```diff
@@ -40,34 +41,44 @@ class DeclarationHolderChangeController @Inject()(
   journeyType: JourneyAction,
   override val exportsCacheService: ExportsCacheService,
   navigator: Navigator,
+  errorHandler: ErrorHandler,
   mcc: MessagesControllerComponents,
   declarationHolderChangePage: declaration_holder_change
 )(implicit ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport with ModelCacheable with SubmissionErrors {
 
-  def displayPage(mode: Mode, id: String): Action[AnyContent] = (authenticate andThen journeyType) { implicit request =>
-    val holder = DeclarationHolderAdd.fromId(id)
-    Ok(declarationHolderChangePage(mode, id, form.fill(holder).withSubmissionErrors()))
+  def displayPage(mode: Mode, id: String): Action[AnyContent] = (authenticate andThen journeyType).async { implicit request =>
+    val maybeExistingHolder = cachedHolders.find(_.id.equals(id))
+
+    maybeExistingHolder.fold(errorHandler.displayErrorPage()) { holder =>
```

Comment:
Just wondering, when there's no an existing declaration holder, what information will be shown to the user when landing on this page.

Reply:
Will the default error page enough to explain what's happening?

Reply:
If the id passed does not match a holder in the cache then `errorHandler.displayErrorPage()` is called which is a generic error page for the service.

Reply:
I got this :+1: . Just wondering if we need to show a more specific error message than the generic "There is a problem. Try again later".

Not a big deal btw.

Reply:
Yeah, its an odd one as this scenario should never be experienced by a user apart from in some very edge cases (like user editing URL param, user bookmarking old page or someone concurrently removing this holder and then trying to edit/remove it on another tab).